### PR TITLE
Add OpenAPI documentation for API v1.0 user and profile endpoints

### PR DESCRIPTION
**Pull request by lyzadanger**
This simple PR adds some more OpenAPI documentation for user and profile endpoints. No user-visible changes here, just part of https://github.com/hypothesis/product-backlog/issues/946

### CODE REVIEW

**Comment by robertknight** [comment]
Testing this with `make docs` was a confusing experience. The docs are compiled into `.tox/py36-docs/html/api-reference` locally. The files in that directory are not updated when the sources in `docs/_extra/api-reference` were changed however. Perhaps you might know what is going on @seanh?

The workaround I used locally was to navigate into `docs/_extra/api-reference`, run a local HTTP server using `python3 -m http.server 8000` and then open http://localhost:8000 in the browser.

**Comment by robertknight** [comment]
I think I might have mentioned this before, but the "Any of: Object | Object" UI looks weird:

screenshot 2019-03-06 11 15 45

It could just be that this is ReDoc's non-ideal rendering. Is there any extra metadata that we can provide to get more helpful names here?

File: docs/_extra/api-reference/hypothesis-v1.yaml [hunk 1]
```diff
@@ -303,3 +320,116 @@ paths:
...
+        '200':
+          description: Success
+          content:
+            application/*json:
+              schema:
+                $ref: '#/components/schemas/Profile'
+
+  /profile/groups:
+    # -----------------------------------------------------
+    # GET user's groups - Retrieve user profile
```
Comment (robertknight) [inline docs/_extra/api-reference/hypothesis-v1.yaml]: Is "Retrieve user profile" correct here in the comment? Do we actually need this comment given that it repeats information in the metadata below?

**Comment by seanh** [comment]
The failure to detect source file changes and update the built docs is probably to do with that our API docs use Swagger hacked into a Sphinx site, rather than properly rendering the API docs in Sphinx.

There are Sphinx extensions like [httpdomain](https://sphinxcontrib-httpdomain.readthedocs.io/en/stable/), [openapi](https://sphinxcontrib-openapi.readthedocs.io/), etc and we should probably be using one of those rather than [this hack](https://github.com/hypothesis/h/blob/c3b607ded27cecbae4b57dad6a053ba3e267980a/docs/_extra/api-reference/index.html#L8-L9). That would hopefully also fix our docs site's navigation being completely missing from the API reference page too.

**Pull request merged**